### PR TITLE
Add function 'nvim-treesitter/install'.ensure_installed

### PR DESCRIPTION
**Pull request by theHamsta**
Hello,

I was trying to create a function that users could put into their config to ensure that their favorite languages are parsers installed or get installed if not yet available.

Currently, you can do:
```
require'nvim-treesitter/install'.ensure_installed({'html', 'python'})
```
or just one language
```
require'nvim-treesitter/install'.ensure_installed('cpp')
```
or all available languages
```
require'nvim-treesitter/install'.ensure_installed('all')
```
I am not sure where to put this function in your repo structure and whether really all requested parsers get installed (the status only shows when a particular languages got installed). So I am looking forward for feedback!

### CODE REVIEW

File: lua/nvim-treesitter/install.lua [hunk 1]
```diff
@@ -143,6 +143,25 @@ local function install(ft)
...
+  if type(languages) == 'string' then
+    if languages == 'all' then
+      languages = configs.available_parsers()
+    else
+      languages = {languages}
+    end
+  end
+
+  for _, ft in ipairs(languages) do
+    local is_installed = #api.nvim_get_runtime_file('parser/'..ft..'.so', false) > 0
```
Comment (kyazdani42) [inline lua/nvim-treesitter/install.lua:157]: you could use the `has_parser` function

**Comment by theHamsta** [comment]
Like so:

```lua
  require'nvim-treesitter.configs'.setup {
      highlight = {
          enable = true,             -- false will disable the whole extension
          disable = {}, -- list of language that will be disabled
      },
      textobj = {                        -- this enables incremental selection
          enable = true,
          disable = {},
      },
      ensure_installed = 'all' -- or {'python', 'cpp', 'rust'}
  }
```
Maybe rename it to `ensure_parser_installed`?

I also fixed a small bug in `iter_cmd` that was using a global which prevented installing multiple parsers in parallel.

File: lua/nvim-treesitter/configs.lua [hunk 2]
```diff
@@ -166,6 +166,7 @@ local config = {
     },
     is_supported = function() return true end
   },
+  ensure_installed = {},
```
Comment (kyazdani42) [inline lua/nvim-treesitter/configs.lua:169]: config keys are actually used by `:TSModuleInfo`, maybe we should rename it to 'modules', i'll do that in a PR. So you can remove ensure_installed from here.

File: lua/nvim-treesitter/install.lua [hunk 3]
```diff
@@ -39,9 +40,11 @@ local function iter_cmd(cmd_list, i, ft)
   local attr = cmd_list[i]
   if attr.info then print(attr.info) end
 
+  local handle
+
```
Comment (kyazdani42) [inline lua/nvim-treesitter/install.lua:44]: cool thanks.

File: lua/nvim-treesitter/configs.lua [hunk 4]
```diff
@@ -325,6 +326,10 @@ function M.setup(user_data)
       if config[mod].keymaps and type(data.keymaps) == 'table' then
         config[mod].keymaps = data.keymaps
       end
+      if mod == 'ensure_installed' then
+        config.ensure_installed = data
+        require'nvim-treesitter/install'.ensure_installed(data)
+      end
```
Comment (kyazdani42) [inline lua/nvim-treesitter/configs.lua:332]: i think as this function will only run once, we don't need to add it to `config.ensure_installed`.
Though i wonder if you had something in mind by storing this in the config object ?

Comment (theHamsta) [reply lua/nvim-treesitter/configs.lua:332]: not, really. I only thought it would be strange if all other data gets stored and this one not.

Comment (kyazdani42) [reply lua/nvim-treesitter/configs.lua:332]: hum well remove it for now, i'll do a small refactoring to make sure this can be stored properly.

Comment (theHamsta) [reply lua/nvim-treesitter/configs.lua:330]: ok. it's removed.

**Pull request merged**